packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -34,6 +34,223 @@ export interface HttpRouteMatch {
+ * Configuration for `HeaderMatch`
+ */
+export interface HeaderMatchConfig {
+  /**
+   * The Http header name
+   */
+  readonly name: string;
+
+  /**
+   * Invert the matching condition.
+   *
+   * @default false
+   */
+  readonly invert?: boolean;
+
+  /**
+   * The match property.
+   */
+  readonly matchProperty: CfnRoute.HeaderMatchMethodProperty;
+}
+
+/**
+ * Used to generate header matching methods.
+ */
+export abstract class HeaderMatch {
+  /**
+   * The value sent by the client must match the specified value exactly.
+   */
+  static valueIs(name: string, exact: string): HeaderMatch {
+    return new HeaderMatchImpl(name, false, { exact });
+  }
+
+  /**
+   * The value sent by the client must not match the specified value exactly.
+   */
+  static valueIsNot(name: string, exact: string): HeaderMatch {
+    return new HeaderMatchImpl(name, true, { exact });
+  }
+
+  /**
+   * The value sent by the client must start with the specified characters.
+   */
+  static valueStartsWith(name: string, prefix: string): HeaderMatch {
+    return new HeaderMatchImpl(name, false, { prefix });
+  }
+
+  /**
+   * The value sent by the client must not start with the specified characters.
+   */
+  static valueDoesNotStartWith(name: string, prefix: string): HeaderMatch {
+    return new HeaderMatchImpl(name, true, { prefix });
+  }
+
+  /**
+   * The value sent by the client must end with the specified characters.
+   */
+  static valueEndsWith(name: string, suffix: string): HeaderMatch {
+    return new HeaderMatchImpl(name, false, { suffix });
+  }
+
+  /**
+   * The value sent by the client must not end with the specified characters.
+   */
+  static valueDoesNotEndWith(name: string, suffix: string): HeaderMatch {
+    return new HeaderMatchImpl(name, true, { suffix });
+  }
+
+  /**
+   * The value sent by the client must include the specified characters.
+   */
+  static valueMatchesRegex(name: string, regex: string): HeaderMatch {
+    return new HeaderMatchImpl(name, false, { regex });
+  }
+
+  /**
+   * The value sent by the client must not include the specified characters.
+   */
+  static valueDoesNotMatchRegex(name: string, regex: string): HeaderMatch {
+    return new HeaderMatchImpl(name, true, { regex });
+  }
+
+  /**
+   * The value sent by the client must be in a range of values
+   * @param start Match on values starting at and including this value
+   * @param end Match on values up to but not including this value
+   */
+  static valuesIsInRange(name: string, start: number, end: number): HeaderMatch {
+    return new HeaderMatchImpl(name, false, {
+      range: {
+        start,
+        end,
+      },
+    });
+  }
+
+  /**
+   * The value sent by the client must not be in a range of values
+   * @param start Match on values starting at and including this value
+   * @param end Match on values up to but not including this value
+   */
+  static valuesIsNotInRange(name: string, start: number, end: number): HeaderMatch {
```

Comment:
I feel like `name` should be `headerName` everywhere here.

That parameter is also missing the JSDocs for each of those methods.